File: 3142/solution.py
class Solution:
    def satisfiesConditions(self, grid: List[List[int]]) -> bool:
        def is_equal_below(i, j):
            if i == rows - 1:
                return True
            return grid[i][j] == grid[i + 1][j]

        def is_different_right(i, j):
            if j == cols - 1:
                return True
            return grid[i][j] != grid[i][j + 1]

        rows = len(grid)
        cols = len(grid[0])
        for i in range(rows):
            for j in range(cols):
                if not is_equal_below(i, j):
                    return False
                if not is_different_right(i, j):
                    return False
        return True

File: 3142/solution.rs
impl Solution {
    pub fn satisfies_conditions(grid: Vec<Vec<i32>>) -> bool {
        let rows = grid.len();
        let cols = if rows > 0 { grid[0].len() } else { 0 };
        for i in 0..rows {
            for j in 0..cols {
                if j < cols - 1 && grid[i][j] == grid[i][j + 1] {
                    return false;
                }
                if i < rows - 1 && grid[i][j] != grid[i + 1][j] {
                    return false;
                }
            }
        }
        true
    }
}
